src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -628,6 +628,19 @@ public void onScopeTreeReady(ScopeTreeReadyEvent event)
          }
       });
       
+      docDisplay_.addEditorBlurHandler((BlurEvent evt) ->
+      {
+         // When the editor loses focus, perform an autosave if enabled, the
+         // buffer is dirty, and we have a file to save to
+         if (prefs.autoSaveOnBlur().getValue() && 
+             dirtyState_.getValue() && 
+             getPath() != null &&
+             !docDisplay_.hasActiveCollabSession())
+         {
+            save();
```

Comment:
Can we do anything to swallow exceptions here, if the save were to fail for some reason?

Reply:
(I'm thinking of the occasional error modal that users sometimes see when trying to work with files in e.g. Dropbox, OneDrive, and so on)

Reply:
👍